### DISCUSSION TITLE
How to use Ctrl-h to delete a character, just like the default behavior in the shell.

### DISCUSSION THREAD

**Discussion by gengdz** [ibhagwan/fzf-lua#1284] [Q&A] [answered]
Currently, when I use C-h, it closes the floating window instead of deleting a character.

![image](https://github.com/ibhagwan/fzf-lua/assets/41912568/29e3e5c4-9235-4d39-8e24-84524f88ac1b)

**Comment by ibhagwan** [comment]
Set ctrl-h in `keymap.fzf` to the appropriate action, I think it’s called `backward-delete-char`, lookup man fzf to be sure.

**Reply by gengdz** [reply]
I read the documentation and found out that it is `backward-delete-char`  after trying.     
 
However, I've noticed that when I use Ctrl-h, it still closes the window directly, while using Ctrl-w works as expected. I suspect that Ctrl-h might have been assigned a different meaning, how should I go about investigating this?

```lua
["ctrl-h"] = "backward-delete-char" -- not working

["ctrl-w"] = "backward-delete-char" -- it works
```

**Reply by gengdz** [reply]
I found out it's not caused by `fzf-lua`, maybe by my `tmux` settings and the `christoomey/vim-tmux-navigator` settings. I need to check into this. 

However, what's strange is that previously in `telescope` I was able to use `Ctrl-h` to delete characters without this issue. 

Is there a concept of priority when it comes to keyboard shortcuts? I don't know.

**Reply by ibhagwan** [reply]
> found out it's not caused by fzf-lua, maybe by my tmux settings and the christoomey/vim-tmux-navigator settings. I need to check into this.

You’re correct that’s the cause as it uses ctrl-h to move window left.

> However, what's strange is that previously in telescope, I was able to use Ctrl-h to delete characters without this issue.

It is not strange, perhaps tmux navigator is limited to term windows (which fzf is)? I don’t know, you’ll have to investigate it yourself and perhaps have an exclusion in the tmux navigator keybind for fzf windows.

I personally use my own cooked version of tmux navigator (written in lua) using the `alt-hjkl` binds, if you wanna take a look:
https://github.com/ibhagwan/nvim-lua/blob/d6a3dd1de64414b48231d6498b901f8c5a8bc9d3/lua/utils.lua#L373

Which then mapped in:
https://github.com/ibhagwan/nvim-lua/blob/d6a3dd1de64414b48231d6498b901f8c5a8bc9d3/lua/keymaps.lua#L66-L71

And in my tmux.conf:
https://github.com/ibhagwan/dots/blob/015a37fd6c402810236f11626085960fb73e101b/.config/tmux/tmux.conf#L70-L78

**Reply by ibhagwan** [reply, marked as answer]
This didn’t happen in telescope because telescope maps `<C-h>` on telescope windows overriding the tmux navigator binds, FzfLua’s ctrl-h is a shell mapping and doesn’t exist on the fzf window so tmux navigator takes over, what you can do is map ctrl-h to itself upon fzf window creation, this should solve your issue:
```lua
    vim.api.nvim_create_autocmd("FileType", {
      group = vim.api.nvim_create_augroup('FzfCtrlH', { clear = true }),
      pattern = "fzf",
      callback = function(e)
        vim.keymap.set("t", "<C-h>", "<C-h>", { buffer = e.buf, silent = true })
      end,
    })
```

**Reply by ibhagwan** [reply]
Btw, I still prefer my own tmux navigator, it also has the cool mapping of `Alt-o` to go back to the last tmux pane similar to `<tmux-prefix>o`, and since I mostly left/right pane this is perfect, if I have a neovim split open it will act as `<C-w>w`.
